src/scripts/authHelper.js

```diff
@@ -1,68 +1,86 @@
 "use strict";
 
 class AuthHelper {
-  _validateDomain(email) {
-    if (email) {
-      return email.includes("@boweryvaluation.com");
-    } else {
-      return false;
-    }
+  constructor() {
+    this._authAttempts = 0;
   }
 
   async runAuthProcess() {
-    const email =
-      (await this._getEmailFromProfile()) ||
-      (await this._getEmailFromStorage()) ||
-      (await this._getEmailFromGoogleOAuth());
-
-    if (email) {
-      await this._saveEmail(email);
-      chrome.browserAction.setPopup({ popup: "../index.html" });
+    this._authAttempts++;
+    if (this._authAttempts > 3) return;
```

Comment:
What should we do if we run out of attempts? It seems that we are doomed in this case.